sage-ios/SAGE/SAGE/CheckinViewController.swift

```diff
@@ -0,0 +1,263 @@
+                    alertController.addAction(UIAlertAction(title: "Start Session", style: .Default, handler: { (action: UIAlertAction) -> Void in
+                        //save start time locally and start timer
+                        self.presentSessionMode(UIConstants.normalAnimationTime)
+                        self.inSession = true
+                        self.startTime = NSDate.timeIntervalSinceReferenceDate()
+                        self.updateSessionTime()
+                        KeychainWrapper.setString(String(format: "%f", self.startTime), forKey: KeychainConstants.kSessionStartTime)
+                    }))
+                    alertController.addAction(UIAlertAction(title: "Cancel", style: .Cancel, handler: nil))
+                    self.presentViewController(alertController, animated: true, completion: nil)
+                } else {
+                    let alertController = UIAlertController(
+                        title: "Location too far",
+                        message: "You are too far from your school. Please move closer to your school to check in.",
+                        preferredStyle: .Alert)
+                    alertController.addAction(UIAlertAction(title: "OK", style: .Default, handler: nil))
+                    self.presentViewController(alertController, animated: true, completion: nil)
+                }
+                break
+            case .Denied, .Restricted, .NotDetermined:
+                self.presentNeedsLocationAlert()
+                break
+            }
+        }
+    }
+    
+    @objc private func userPressedEndSession(sender: UIButton!) {
+        //Verify location
+        if true {
+            let alertController = UIAlertController(
+                title: "Finish mentoring session?",
+                message: nil,
+                preferredStyle: .ActionSheet)
+            alertController.addAction(UIAlertAction(title: "Finish Session", style: .Default, handler: { (action: UIAlertAction) -> Void in
+                self.presentRequestHoursView()
+            }))
+            alertController.addAction(UIAlertAction(title: "Cancel Session", style: .Destructive, handler: { (action: UIAlertAction) -> Void in
+                let confirmAlert = UIAlertController(
+                    title: "Are you sure? Your session will be erased.",
+                    message: nil,
+                    preferredStyle: .ActionSheet)
+                confirmAlert.addAction(UIAlertAction(title: "Continue", style: .Default, handler: { (action: UIAlertAction) -> Void in
+                    // send request to make check in and present to default
+                    // when finished. if failure, store locally and
+                    // try again until success
+                    self.presentDefaultMode(UIConstants.normalAnimationTime)
+                    self.inSession = false
+                    KeychainWrapper.removeObjectForKey(KeychainConstants.kSessionStartTime)
```

Comment:
Hm does the keychain stuff work? Because before you were using the setStringForKey method and now you're using the removeObjectForKey  - I was having problems with this. I think you're supposed to use only the string methods or the object methods for a certain key, so making that consistent would be good :)